### ISSUE TITLE
⌨ Table focus order issue

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#8109]
## ⌨ Accessibility Issue

The order of the table filter relative to the "Add a new Item" Button is confusing. While navigating the Skills table with a reader, a user could be confused as to why they can search for skills before they are asked to add a new one. 

### WCAG 2.1 Criteria

List any relevant criteria that is failing.

## 🦋 Expected Behaviour

Users should be able to navigate from the filters directly to the table content they are filtering without having to tab through a button that changes their entire context. 

## 🕵️ Details

- **Operating System**: Windows, macOS, Linux
- **Browser**: Chrome, Edge, Safari, Firefox
- **Assistive Technology**: NVDA, Narrator, VoiceOver, Orca

## 📋 Steps to Reproduce

1.  Open the Behavioural skill library with a screen reader.
2.  Search for a skill using the filter. 
3.  Check if your skill has been added to the table. Listen to what the screen reader does on the way into the table contents. 
4.  Go back to the filter to the search for a new skill. 

## 📸 Screenshot

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcVR4IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--2623d1a0656c02481e61cff03ed15cb5f57d2ce3/image.png)

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Criteria 1
- [ ] Criteria 2

**Comment by petertgiles** [comment]
Is the proposed solution to change the visual design at all?  Or it appears the same but tabs differently?

**Comment by RM-1978** [comment]
Ideally tab order follows reading order, but we are happy to defer to the @GCTC-NTGC/designers on whether the visual design changes.

**Comment by substrae** [comment]
@petertgiles 

This is precisely what the CSS property `order` is for - we can position the "add" button first in the DOM and use `order` to position it in the final grid/flex spot in the controls row.

**Comment by petertgiles** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/GC-Digital-Talent-6092dc05157fa30010edaef9/issues/GCTC-NTGC/gc-digital-talent/8109?planning-poker) with Zenhub @brindasasi @esizer @JamesHuf @mnigh @tristan-orourke @vd1992 @yonikid15
